arviz/tests/test_utils.py

```diff
@@ -134,3 +151,83 @@ def placeholder_func():
 )
 def test_format_sig_figs(value, default, expected):
     assert format_sig_figs(value, default=default) == expected
+
+
+def test_conditional_vect_numba_decorator():
+    """Tests to see if Numba is used.
+
+    Test can be distinguished from test_conditional_jit_decorator_no_numba
+    by use of debugger or coverage tool
+    """
+    from arviz import utils
+
+    @utils.conditional_vect
+    def func(a_a, b_b):
+        return a_a + b_b
+
+    value_one = np.random.randn(10)
+    value_two = np.random.randn(10)
+    assert np.allclose(func(value_one, value_two), value_one + value_two)
+
+
+def test_conditional_vect_numba_decorator_keyword(monkeypatch):
+    """Checks else statement and vect keyword argument"""
+    from arviz import utils
+
+    # Mock import lib to return numba with hit method which returns a function that returns kwargs
+    numba_mock = Mock()
+    monkeypatch.setattr(utils.importlib, "import_module", lambda x: numba_mock)
+
+    def vectorize(**kwargs):
+        """overwrite numba.vectorize function"""
+        return lambda x: (x(), kwargs)
+
+    numba_mock.vectorize = vectorize
+
+    @utils.conditional_vect(keyword_argument="A keyword argument")
+    def placeholder_func():
+        """This function does nothing"""
+        return "output"
+
+    # pylint: disable=unpacking-non-sequence
+    function_results, wrapper_result = placeholder_func
+    assert wrapper_result == {"keyword_argument": "A keyword argument"}
+    assert function_results == "output"
+
+
+def test_numba_check():
+    """Test for numba_check"""
+    numba = importlib.util.find_spec("numba")
+    flag = numba is not None
+    assert flag == numba_check()
+
+
+def test_numba_utils():
+    """Test for class Numba."""
+    flag = Numba.numba_flag
+    assert flag == numba_check()
+    Numba.disable_numba()
+    val = Numba.numba_flag
+    assert not val
+    Numba.enable_numba()
+    val = Numba.numba_flag
+    assert val
+    assert flag == Numba.numba_flag
+
+
+@pytest.mark.parametrize("axis", (0, 1))
```

Comment:
@Ban-zee As I get time Im going to keep picking with comments :)

Can we move numba disable to a parametrization? That way if the numba fails it's easier to tell because the tests are separated?

Reply:
Sure